mobile/library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -1,17 +1,20 @@
 package io.envoyproxy.envoymobile.engine;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.ArrayList;
 import java.util.Map;
 import java.util.regex.Pattern;
 import java.util.regex.Matcher;
+import java.util.StringJoiner;
```

Comment:
I don't think StringJoiner is available to all the OS versions we currently support: https://github.com/envoyproxy/envoy-mobile/pull/2098

In my research, I haven't found a way to detect this at compile time, which is a bit baffling to me.